src/GitHub.App/Infrastructure/LogManager.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.IO;
+using GitHub.Extensions;
+using Serilog;
+using Serilog.Core;
+
+namespace GitHub.Infrastructure
+{
+    public static class LogManager
+    {
+        static Logger CreateLogger()
+        {
+            var environment = new Rothko.Environment();
+            var logPath = Path.Combine(environment.GetLocalGitHubApplicationDataPath(), "extension.log");
+
+            const string outputTemplate = "{Timestamp:yyyy-MM-dd HH:mm:ss.fff}|{Level}|Thread:{ThreadId}|{SourceContext}|{Message}{NewLine}{Exception}";
+
+            //2MBs
+            const long fileSizeLimitBytes = 2L*1024L*1024L;
+
+            return new LoggerConfiguration()
+                .Enrich.WithThreadId()
+                .WriteTo.File(logPath,
+                    fileSizeLimitBytes: fileSizeLimitBytes,
+                    outputTemplate: outputTemplate)
+                .CreateLogger();
+        }
+
+        static Lazy<Logger> Logger { get; } = new Lazy<Logger>(CreateLogger);
+
+        public static ILogger ForContext<T>() => Logger.Value.ForContext<T>();
```

Comment:
So a function `FooBar<T>()` that does not return something with type `T` is apparently a code analysis error.
It can't disabled with a `#pragma`. It would have to be modified out of the rule set.

https://msdn.microsoft.com/library/ms182150.aspx